include/xtensor/xtensor.hpp

```diff
@@ -518,6 +518,31 @@ namespace xt
     {
         return m_storage;
     }
+
+    /**
+     * Converts `std::vector<index_type>` (return e.g. from `xt::argwhere`) to a `xt::tensor<size_t,2>`
+     * @param vector of indices
+     */
+    template <class T>
+    inline auto from_indices(const std::vector<T> &idx)
+    {
+        if ( idx.size() == 0 )
+        {
+            return empty<std::size_t>({0,0});
+        }
+
+        xtensor<std::size_t,2> out = empty<std::size_t>({idx.size(),idx[0].size()});
+
+        for ( std::size_t i = 0 ; i < out.shape()[0] ; ++i )
+        {
+            for ( std::size_t j = 0 ; j < out.shape()[1] ; ++j )
+            {
+                out(i,j) = static_cast<std::size_t>(idx[i][j]);
+            }
+        }
+
+        return out;
```

Comment:
I think we can keep `T` instead of forcing a conversion to `size_t` so this functino can work with signed integers or even circular indices, and the type is not lost when converting to the tensor.

Reply:
I hope too, but then you have to help me a little but to undress `T`:

```cpp
 error: assigning to 'unsigned long' from incompatible type 'std::__1::array<unsigned long, 1>'
```

Reply:
Oops, that should be `typename T::value_type`

Reply:
Thanks! Resolved.